### ISSUE TITLE
Plugin Popup is often hidden by Core popup

### ISSUE THREAD

**Issue by Klumbumbus** [kmpoppe/noteSolver#3]
After clicking on a note in the mapview two poups appear which often overlap each other.

See screenshot.

![grafik](https://user-images.githubusercontent.com/7115609/84167628-c79c6600-aa76-11ea-8760-f67a9e834c25.png)

````
URL:https://josm.openstreetmap.de/svn/trunk
Repository:UUID: 0c6e7542-c601-0410-84e7-c038aed88b3b
Last:Changed Date: 2020-06-08 23:23:28 +0200 (Mon, 08 Jun 2020)
Build-Date:2020-06-09 01:30:47
Revision:16584
Relative:URL: ^/trunk

Identification: JOSM/1.5 (16584 de) Windows 10 64-Bit
OS Build number: Windows 10 Pro 1909 (18363)
Memory Usage: 686 MB / 1820 MB (181 MB allocated, but free)
Java version: 1.8.0_251-b08, Oracle Corporation, Java HotSpot(TM) 64-Bit Server VM
Look and Feel: com.sun.java.swing.plaf.windows.WindowsLookAndFeel
Screen: \Display0 1680x1050 (scaling 1.0x1.0)
Maximum Screen Size: 1680x1050
Best cursor sizes: 16x16 -> 32x32, 32x32 -> 32x32
VM arguments: [-Djava.security.manager, -Djava.security.policy=file:<java.home>\lib\security\javaws.policy, -DtrustProxy=true, -Djnlpx.home=<java.home>\bin, -Djnlpx.origFilenameArg=C:\Program Files (x86)\josm-latest.jnlp, -Djnlpx.remove=false, -Djava.util.Arrays.useLegacyMergeSort=true, -Djnlpx.heapsize=NULL,2048m, -Djnlpx.splashport=63381, -Djnlpx.jvm=<java.home>\bin\javaw.exe]
Dataset consistency test: No problems found

Plugins:
+ DirectUpload (35248)
+ HouseNumberTaggingTool (35248)
+ OpeningHoursEditor (35414)
+ PicLayer (35405)
+ RoadSigns (35424)
+ SimplifyArea (35405)
+ apache-commons (35362)
+ apache-http (35092)
+ buildings_tools (35474)
+ editgpx (35248)
+ ejml (35313)
+ imagery-xml-bounds (35313)
+ imagery_offset_db (35405)
+ javafx-windows (35375)
+ jaxb (35092)
+ jna (35092)
+ jogl (1.2.3)
+ log4j (35092)
+ measurement (35405)
+ notesolver (0.3.2)
+ photo_geotagging (35405)
+ photoadjust (35405)
+ reltoolbox (35405)
+ reverter (35487)
+ rex (53)
+ tageditor (35258)
+ tagging-preset-tester (35267)
+ terracer (35327)
+ turnlanes-tagging (283)
+ turnrestrictions (35405)
+ undelete (35474)
+ utilsplugin2 (35476)
````

**Comment by Klumbumbus** [comment]
Also when you click on a note in the notes panel on the right the popup appears there too. It would be better if it only appears there on right click.

**Comment by kmpoppe** [comment]
Thanks @Klumbumbus, you're making two very good points there :-)

Unfortunately, I hadn't been able to find either a way to force the pop-up to be the top most layer or a way to only react to a right click mouse event. 

I had checked to online documentation and other Plugins for any examples. 

Could you, as a @josm member put me into contact with someone who could know how to solve these shortcomings?

K.

**Comment by Klumbumbus** [comment]
You can ask on josm-dev mailing list

**Comment by simon04** [comment]
> or a way to only react to a right click mouse event

Use `javax.swing.SwingUtilities#isRightMouseButton`

**Comment by simon04** [comment]
You could take a look at `org.openstreetmap.josm.gui.layer.AbstractTileSourceLayer#adapter`, i.e., the popup menu for imagery layers.

**Comment by kmpoppe** [comment]
This might also be solved by #21, will check.

**Comment by ygra** [comment]
This will be solved, although by modifying as little code as possible. It won't integrate better into JOSM, but at least the menu won't be hidden by the other popup anymore.

**Comment by ygra** [comment]
> Also when you click on a note in the notes panel on the right the popup appears there too. It would be better if it only appears there on right click.

There is a context menu for notes now in JOSM; alas it isn't extensible by plugins.

**Comment by kmpoppe** [comment]
> There is a context menu for notes now in JOSM; alas it isn't extensible by plugins.

I know, I was also in discussions with one of the dev to make it extensible, but there hasn't been a solution as of yet.

**Comment by kmpoppe** [comment]
Resolved by 7e97eec
